bincopy.py

```diff
@@ -591,7 +591,8 @@ def as_binary(self, minimum_address=None, padding=b'\xff'):
         :param minimum_address: Start address of the resulting binary data. Must
                         be less than or equal to the start address of
                         the binary data.
-        :param padding: Value of the padding between not adjecent segments.
+        :param padding: byte value of the padding between not adjecent segments.
+        :           padding byte will be added of word_size_bytes times
```

Comment:
I think it's better to have the padding in words to make it possible to pad with for example `\xff\x00`. With byte padding only `\xff\xff` or `\x00\x00` would be possible.

```
        :param padding: Word value of the padding between non-adjecent segments.
                        Give as a bytes object of length 1 when the word size is 8 bits, length 2 when
                        the word size is 16 bits, and so on.
```

The implementation would change to:

```

    def as_binary(self, minimum_address=None, padding=None):
        """Return a byte string of all data.

        :param minimum_address: Start address of the resulting binary data. Must
                        be less than or equal to the start address of
                        the binary data.
        :param padding: Word value of the padding between non-adjecent segments.
                        Give as a bytes object of length 1 when the word size is 8 bits, length 2 when
                        the word size is 16 bits, and so on.
        :returns: A byte string of the binary data.

        """

        if self.segments.get_size() == 0:
            return b''

        res = b''
        current_maximum_address = self.get_minimum_address()

        if padding is None:
            padding = b'\xff' * self.word_size_bytes
        
        if minimum_address is not None:
            if minimum_address > self.get_minimum_address():
                raise Error('the selected start address must be lower or '
                            'equal to the start address of the binary')

            current_maximum_address = minimum_address

        for address, data in self.segments.iter():
            address //= self.word_size_bytes
            res += padding * (address - current_maximum_address)
            res += data
            current_maximum_address = address + (len(data) // self.word_size_bytes)

        return res

    def as_array(self, minimum_address=None, padding=None, separator=', '):
```

Note `padding=None` in the `as_array()` function at the end.